src/circuit/gadget/utilities/cond_swap.rs

```diff
@@ -176,18 +176,17 @@ impl<F: FieldExt> CondSwapChip<F> {
             let b_swapped = meta.query_advice(config.b_swapped, Rotation::cur());
             let swap = meta.query_advice(config.swap, Rotation::cur());
 
-            let one = Expression::Constant(F::one());
-
-            // a_swapped - b ⋅ swap - a ⋅ (1-swap) = 0
             // This checks that `a_swapped` is equal to `b` when `swap` is set,
             // but remains as `a` when `swap` is not set.
-            let a_check =
-                a_swapped - b.clone() * swap.clone() - a.clone() * (one.clone() - swap.clone());
+            let a_check = ternary(
+                swap.clone(),
+                a_swapped.clone() - b.clone(),
+                a_swapped - a.clone(),
+            );
```

Comment:
Hmm, this is a larger change than I'd anticipated. I had expected this to look like:
```suggestion
            let a_check = a_swapped - ternary(swap.clone(), b.clone(), a.clone());
```
which would result in a circuit change that is just rearranging some negations. This approach works because `ternary` returns an expression where the multiplicative factor of each half is either `0` or `1`, not `-1`.

Weakly blocking; I'm also fine with this approach (it results in a slightly larger constraint but that is clearly correct regardless of how `ternary` works), but I'd prefer if it were implemented across two commits, to make review easier (first commit introducing `ternary` per my suggestion above, second commit switching to the encompassing usage).